sdk-python/livescoreapi.py

```diff
@@ -7,17 +7,62 @@
 """
 
 import urllib.request
-
+import json
 
 class livescores:
+    """
+    A class used to get livescores. 
+    ...
+
+    Attributes
+    ----------
+    key : str
+        an alphanumeric string used to access data from the LiveScore API
+    secret : str
+        an alphanumeric string used to access data from the LiveScore API
     
-    def get_all(key,secret):
-        html_file = urllib.request.urlopen("http://livescore-api.com/api-client/scores/live.json?key=" + key + "&secret=" + secret)
-        json_data = html_file.read()
-        return json_data
+    Methods
+    -------
+    _get_html_file(link)
     
-
+    get_all()
+    
+    """
+    link_get_all = "http://livescore-api.com/api-client/scores/live.json?key={}&secret={}"
+   
+    def __init__(self, key, secret):
         
-                
+        try: 
+            key.strip()
+            secret.strip()
+            assert len(key) == 16
+            assert len(secret) == 32
+            self.key = key
+            self.secret = secret 
+            
+        except Exception as e:
+            print("Key or Secret is Empty!")
     
-
+    def _make_api_call(self, link):
+        """returns the source code given a link"""
+    
+        try:
+            http_response_object = urllib.request.urlopen(link)
+            if http_response_object is not None: 
+                json_string = http_response_object.read().decode('utf-8')
+                json_dic = json.loads(json_string)
+                return json_dic
+            
+        except urllib.error.HTTPError as e:
+            print (str(e))
+        
+        except urllib.error.URLError as e:
+            print (str(e))
+        
+        except urllib.error.ContentTooShortError as e:
+            print (str(e))
+            
+    def get_all(self):
+        """returns all the livescores"""
+        link = self.link_get_all.format(self.key,self.secret)
```

Comment:
All api calls include a api key and api secret this should be put into a separate function so we do not have to write it every time. Keep in mind that when it comes to URLs we have different URLs with different parameters so we need to be flexible here.